library/java/net/openid/appauth/AuthorizationService.java

```diff
@@ -495,11 +504,111 @@ protected void onPostExecute(JSONObject json) {
                 return;
             }
 
+            if (response.idToken != null) {
+                IdToken idToken;
```

Comment:
Could you move all of this validation logic in a utility method within the IdToken class? I think that will help to keep all of the parsing and validation logic in one place.